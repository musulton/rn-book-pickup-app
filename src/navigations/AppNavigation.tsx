import React from "react";
import {createNativeStackNavigator} from "@react-navigation/native-stack";
import {NavigationContainer, DefaultTheme} from "@react-navigation/native";

import TabNavigation from "./TabNavigation";
import BookDetails from "../screens/BookDetails";
import {ROUTES} from "../constants";

const Stack = createNativeStackNavigator();

const Theme = {
    ...DefaultTheme,
    colors: {
        ...DefaultTheme.colors,
        border: "transparent"
    }
}

const AppNavigation: React.FC = () => {
    return (
        <NavigationContainer theme={Theme}>
            <Stack.Navigator
                initialRouteName={ROUTES.HOME}
            >
                <Stack.Screen
                    name={ROUTES.HOME}
                    component={TabNavigation}
                    options={{
                        headerShown: false
                    }}
                />
                <Stack.Screen
                    name={ROUTES.BOOK_DETAILS}
                    component={BookDetails}
                    options={{
                        title: "Book Details"
                    }}
                />
            </Stack.Navigator>
        </NavigationContainer>
    )
}

export default AppNavigation
