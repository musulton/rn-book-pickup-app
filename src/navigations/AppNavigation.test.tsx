import React from "react";
import {render, screen} from "@testing-library/react-native";

import AppNavigation from "./AppNavigation";

describe("<AppNavigation />", () => {
    it("renders <AppNavigation /> correctly", () => {
        render(<AppNavigation />)

        expect(screen.toJSON()).toMatchSnapshot()
    })
})
