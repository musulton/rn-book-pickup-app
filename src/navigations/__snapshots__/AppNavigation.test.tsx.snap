// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavigation /> renders <AppNavigation /> correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <RNSScreenNavigationContainer
              style={
                {
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <RNSScreen
                activityState={2}
                collapsable={false}
                forwardedRef={[Function]}
                gestureResponseDistance={
                  {
                    "bottom": -1,
                    "end": -1,
                    "start": -1,
                    "top": -1,
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 0,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  importantForAccessibility="auto"
                  style={
                    [
                      {
                        "backgroundColor": "rgb(242, 242, 242)",
                        "flex": 1,
                      },
                      [
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      style={
                        {
                          "backgroundColor": "#FEFEFE",
                          "paddingTop": 20,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "borderBottomColor": "#EFEFF0",
                            "borderBottomWidth": 1,
                            "display": "flex",
                            "justifyContent": "flex-start",
                            "padding": 20,
                          }
                        }
                      >
                        <Text
                          style={
                            {
                              "color": "#64676D",
                            }
                          }
                        >
                          Hi, Your account name
                        </Text>
                        <Text
                          style={
                            {
                              "fontSize": 20,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Discover Latest Book
                        </Text>
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#EFEFF0",
                              "borderRadius": 6,
                              "flexDirection": "row",
                              "marginTop": 10,
                              "padding": 10,
                            }
                          }
                        >
                          <TextInput
                            onBlur={[Function]}
                            onChangeText={[Function]}
                            onFocus={[Function]}
                            placeholder="Search book.."
                            style={
                              {
                                "fontSize": 20,
                                "marginLeft": 10,
                                "width": "90%",
                              }
                            }
                            value=""
                          />
                        </View>
                      </View>
                      <RCTScrollView
                        ListEmptyComponent={<EmptyState />}
                        ListFooterComponent={null}
                        contentContainerStyle={
                          {
                            "backgroundColor": "#FEFEFE",
                            "display": "flex",
                            "flex": 1,
                          }
                        }
                        data={[]}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        initialNumToRender={10}
                        keyExtractor={[Function]}
                        onContentSizeChange={[Function]}
                        onEndReached={[Function]}
                        onLayout={[Function]}
                        onMomentumScrollBegin={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        refreshing={true}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={[]}
                        testID="bookList"
                        viewabilityConfigCallbackPairs={[]}
                      >
                        <View>
                          <View
                            style={
                              {
                                "alignItems": "center",
                                "display": "flex",
                                "flex": 1,
                                "justifyContent": "center",
                              }
                            }
                          >
                            <Text>
                              Data is nof found
                            </Text>
                          </View>
                        </View>
                      </RCTScrollView>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </RNSScreen>
            </RNSScreenNavigationContainer>
            <View
              collapsable={false}
              onLayout={[Function]}
              pointerEvents="auto"
              style={
                {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderTopColor": "#EFEFF0",
                  "borderTopWidth": 1,
                  "bottom": 0,
                  "elevation": 8,
                  "height": 49,
                  "left": 0,
                  "paddingBottom": 0,
                  "paddingHorizontal": 0,
                  "position": null,
                  "right": 0,
                  "transform": [
                    {
                      "translateY": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                accessibilityRole="tablist"
                style={
                  {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityLabel="books, tab, 1 of 2"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": true,
                    }
                  }
                  accessibilityStates={
                    [
                      "selected",
                    ]
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
                <View
                  accessibilityLabel="history, tab, 2 of 2"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": false,
                    }
                  }
                  accessibilityStates={[]}
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
